### ISSUE TITLE
[java] UnusedPrivateMethod with method ref

### ISSUE THREAD

**Issue by patpatpat123** [pmd/pmd#5621]
**Affects PMD Version:** 7.7.0

**Rule:** [UnusedPrivateMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**

**Code Sample demonstrating the issue:**

```java

    private String foo(final Object foo, final String bar) {
        final String abc = Optional.ofNullable(foo.toString()).map(Foo::getNormalizedAddress).orElse(null);
        return abc;
    }

private static String getNormalizedAddress(final String encoded) {
        try (var objectInputStream = new ObjectInputStream(new ByteArrayInputStream(Base64.getDecoder().decode(encoded)))) {
            final var inetSocketAddress = (InetSocketAddress) objectInputStream.readObject();
            return inetSocketAddress.toString().substring(1);
        } catch (IOException | ClassNotFoundException | IllegalArgumentException e) {
            return null;
        }
    }

```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

The method is clearly being used

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Maven

```
 <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-pmd-plugin</artifactId>
                <version>3.26.0</version>
            </plugin>
```

**Comment by adangel** [comment]
@patpatpat123 Can you retry with the latest PMD version?
See https://maven.apache.org/plugins/maven-pmd-plugin/examples/upgrading-PMD-at-runtime.html for how to use the latest PMD version with maven-pmd-plugin.

This could be a duplicate of #5083 (fixed with 7.8.0)

**Comment by pn-santos** [comment]
I'm also still getting the `UnsusedPrivateMethod` (PMD 7.11.0) with code like this:
```java
    @Override
    public Collection<Consumer<Router>> routes() {
        return List.of(
                r -> r.post("/some/path").handler(this::handleRequest)
        );
    }

    private void handleRequest(RoutingContext rc) { ... }
```

**Comment by patpatpat123** [comment]
Hello @adangel ,

Similar to what @pn-santos said, I still get this false positive even with 7.11.0.

Could you please help us?

**Comment by oowekyala** [comment]
I think even though it has not been fixed yet it is related to #5083. I will investigate.

**Comment by oowekyala** [comment]
I cannot reproduce the issue with either code samples you provided. Is there more to the code than what we can see here? Are any of those methods overloaded maybe?

**Comment by pn-santos** [comment]
> Are any of those methods overloaded maybe?

In my case yes:
```java
public interface Resource {

    Collection<Consumer<Router>> routes();

    void applyRoutes(Router router);

}
```
```java
public abstract class AbstractResource implements Resource {

    @Override
    public final void applyRoutes(Router router) {
        routes().forEach(route -> route.accept(router));
    }

}
```
```java
public final class AdminResource extends AbstractResource {

    @Override
    public Collection<Consumer<Router>> routes() {
        return List.of(
                r -> r.post("/some/path").handler(this::handleRequest)
        );
    }

    private void handleRequest(RoutingContext rc) { ... }

}
```

**Comment by pankratz76** [comment]
> I cannot reproduce the issue with either code samples you provided. Is there more to the code than what we can see here? Are any of those methods overloaded maybe?

@oowekyala i can reproduce this locally: https://github.com/pmd/pmd/pull/5659

**Comment by ryleighmikami** [comment]
Same issue here. Is it possible to fix this and bump the plugin to use the latest PMD as well please?
